lib/streamlit/runtime/pages_manager.py

```diff
@@ -15,322 +15,119 @@
-    """
-
-    DefaultStrategy: type[PagesStrategyV1 | PagesStrategyV2] = PagesStrategyV1
-
     def __init__(
         self,
         main_script_path: ScriptPath,
-        script_cache: ScriptCache | None = None,
-        **kwargs,
+        *,
+        # setup_watcher is a convenience flag to allow disabling the watcher for testing
+        setup_watcher: bool = True,
     ):
         self._main_script_path = main_script_path
         self._main_script_hash: PageHash = calc_md5(main_script_path)
-        self.pages_strategy = PagesManager.DefaultStrategy(self, **kwargs)
-        self._script_cache = script_cache
-        self._intended_page_script_hash: PageHash | None = None
-        self._intended_page_name: PageName | None = None
         self._current_page_script_hash: PageHash = ""
+        pages_dir = self.main_script_parent / "pages"
+        self._is_mpa_v1 = os.path.exists(pages_dir)
+        if self._is_mpa_v1:
+            if setup_watcher:
+                source_util.setup_pages_watcher(pages_dir)
+            self._pages = source_util.get_pages(self._main_script_path)
+        else:
+            self._pages = {}
 
-    @property
-    def main_script_path(self) -> ScriptPath:
-        return self._main_script_path
+    def get_pages(self) -> dict[PageHash, PageInfo]:
+        return self._pages
+
+    def set_pages(self, pages: dict[PageHash, PageInfo]):
+        if self._is_mpa_v1:
+            # Log the warning and reset the "strategy" to V2
+            _LOGGER.warning(
+                "st.navigation was called in an app with a pages/ directory. "
+                "This may cause unusual app behavior. You may want to rename the "
+                "pages/ directory."
+            )
+            self._is_mpa_v1 = False
+        self._pages = pages
 
-    @property
-    def main_script_parent(self) -> Path:
-        return Path(self._main_script_path).parent
+    def get_main_page(self) -> PageInfo:
+        return {
+            "script_path": self._main_script_path,
+            "page_script_hash": self._main_script_hash,
+            "url_pathname": "",
+        }
 
-    @property
-    def main_script_hash(self) -> PageHash:
-        return self._main_script_hash
+    def get_page_script_by_hash(self, page_script_hash: PageHash) -> PageInfo | None:
+        if page_script_hash in self._pages:
+            return self._pages[page_script_hash]
+
+        return None
+
+    def get_page_script_by_name(self, page_name: PageName) -> PageInfo | None:
+        for page in self._pages.values():
+            if page["url_pathname"] == page_name:
+                return page
+
+        return None
 
     @property
     def current_page_script_hash(self) -> PageHash:
         return self._current_page_script_hash
 
     @property
-    def intended_page_name(self) -> PageName | None:
-        return self._intended_page_name
+    def mpa_version(self) -> int:
+        return 1 if self._is_mpa_v1 else 2
```

Comment:
nit: I would introduce an enum and use that instead of numbers for cleaner reusability

Reply:
How about a boolean on whether we are in v1 or not? I agree an enum is cleaner, but it adds an import for checking. Since at some point we plan to deprecate v1, it might be an easier fix?